pkg/reportingplugins/mercury/v1/types.go

```diff
@@ -9,14 +10,59 @@ import (
 	"github.com/smartcontractkit/chainlink-relay/pkg/reportingplugins/mercury"
 )
 
+type Block struct {
+	Num  int64
+	Hash string // Hash is stringified to allow use of block as hash key. It is NOT hex
+	Ts   uint64
+}
+
+func NewBlock(num int64, hash []byte, ts uint64) Block {
+	return Block{
+		Num:  num,
+		Hash: string(hash),
+		Ts:   ts,
+	}
+}
+
+// b1 is less than b2 if it is:
+// smaller block number
+// smaller timestamp
+// largest hash
+// evaluated in that order
+func (b Block) less(b2 Block) bool {
+	if b.Num == b2.Num && b.Ts == b2.Ts {
+		// tie-break on hash, all else being equal
+		return b.Hash > b2.Hash
+	} else if b.Num == b2.Num {
+		// if block number is equal and timestamps differ, take the oldest timestamp
+		return b.Ts < b2.Ts
+	} else {
+		// if block number is different, take the lower block number
+		return b.Num < b2.Num
+	}
```

Comment:
The linter usually wants this form:
```suggestion
	}
	// if block number is different, take the lower block number
	return b.Num < b2.Num
```

Reply:
It doesn't seem to have complained about this one though... should I change it?

Reply:
Could leave it to see if it fails on develop as a test. It has been malfunctioning a lot lately